js/clinicalTimeline.js

```diff
@@ -1090,6 +1112,30 @@ var clinicalTimeline = (function(){
     return getOrSetBool("advancedView", b);
   };
 
+  window.handlePlugins = function (plugin, bool) {
+    switch(plugin) {
+      case "trimTimeline":
+        getOrSetBool("advancedView", !bool);
+        getOrSetBool("enableTrimmedTimeline", bool);
+        getOrSetBool("enableVerticalLine", !bool);
+        getOrSetBool("enableZoom", !bool);
+        break;
+      case "verticalLine":
+        getOrSetBool("advancedView", !bool);
+        getOrSetBool("enableTrimmedTimeline", !bool);
+        getOrSetBool("enableVerticalLine", bool);
+        getOrSetBool("enableZoom", !bool);
+        break;
+      case "zoom":
+        getOrSetBool("advancedView", bool);
+        getOrSetBool("enableZoom", bool);
+        break;
+      default:
+        console.log("Plugin " + plugin + " not available");
+    }
+    timeline();
+  }
```

Comment:
This makes it more clear what booleans are used in what plugin, but there are still a lot of variables being used in multiple plugins that are weaving the plugins together instead of separating them. It's exposing an API that is a little more clean because it shows there are just three plugins, but under the hood everything is still connected.